File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/interfaces/DevLoadingViewManager.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.devsupport.interfaces;

import androidx.annotation.Nullable;

/** Interface to display loading messages on top of the screen. */
public interface DevLoadingViewManager {

  void showMessage(final String message);

  void updateProgress(
      final @Nullable String status, final @Nullable Integer done, final @Nullable Integer total);

  void hide();
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/interfaces/DevLoadingViewManager.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.devsupport.interfaces

/** Interface to display loading messages on top of the screen. */
public interface DevLoadingViewManager {
  public fun showMessage(message: String)

  public fun showMessage(message: String, color: Double?, backgroundColor: Double?)

  public fun updateProgress(status: String?, done: Int?, total: Int?)

  public fun hide()
}
